riotfile.py

```diff
@@ -1213,37 +1213,56 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
             ],
         ),
         Venv(
-            # aiobotocore: aiobotocore>=1.0 not yet supported
             name="aiobotocore",
             command="pytest {cmdargs} tests/contrib/aiobotocore",
             pkgs={
                 "pytest-asyncio": latest,
             },
             venvs=[
+                # async_generator 1.10 used because @asynccontextmanager was only available in Python 3.6+
+                # aiobotocore 1.x and higher require Python 3.6 or higher
+                Venv(
+                    pys=select_pys(min_version="3.6"),
+                    pkgs={
+                        "aiobotocore": ["~=2.0.0", "~=2.1.0", "~=2.2.0", "~=2.3.0"],
+                        "async_generator": ["~=1.10"],
+                    },
+                ),
+                Venv(
+                    pys=select_pys(min_version="3.6"),
+                    pkgs={
+                        "aiobotocore": ["~=1.0.0", "~=1.2.0", "~=1.3.0", "~=1.4.2"],
+                        "async_generator": ["~=1.10"],
+                    },
+                ),
                 Venv(
                     pys=select_pys(min_version="3.5", max_version="3.6"),
                     pkgs={
                         "aiobotocore": ["~=0.2", "~=0.3", "~=0.4"],
+                        "async_generator": ["~=1.10"],
                     },
                 ),
                 # aiobotocore 0.2 and 0.4 do not work because they use async as a reserved keyword
                 Venv(
                     pys=select_pys(min_version="3.5", max_version="3.8"),
                     pkgs={
                         "aiobotocore": ["~=0.5", "~=0.7", "~=0.8", "~=0.9"],
+                        "async_generator": ["~=1.10"],
```

Comment:
if this dependency is used in each venv then we can move it to the parent venv on line 1220